src/compile/data/facet.ts

```diff
@@ -75,24 +64,47 @@ export class FacetNode extends DataFlowNode {
     return this.name;
   }
 
-  private assembleRowColumnData(channel: 'row' | 'column', crossedDataName: string): VgData {
-    const childChannel = channel === 'row' ? 'y' : 'x';
+  private getChildIndependentFieldWithStep() {
+    const childIndependentFieldsWithStep: ChildIndependentFieldsWithStep = {};
 
+    for (const channel of ['x', 'y'] as ScaleChannel[]) {
+      const childScaleComponent = this.childModel.component.scales[channel];
+      if (childScaleComponent && !childScaleComponent.merged) {
+        const type = childScaleComponent.get('type');
+        const range = childScaleComponent.get('range');
+
+        if (hasDiscreteDomain(type) && isVgRangeStep(range)) {
+          const domain = assembleDomain(this.childModel, channel);
```

Comment:
Yeah but here we need to know the field of the child (if the child merge domains of its children), and the only way to know the field is from domain, but then domain cannot be merged correctly here because it depends on the side effect that hasn't happened.  

If you don't want to refactor, then tell me how would you fix this problem otherwise?